// slider-carousel/esm5/slider-carousel-preview/slider-carousel-preview.component.js
import { Component, OnInit, ElementRef, HostBinding } from '@angular/core';
var SliderCarouselPreviewComponent = /** @class */ (function () {
    function SliderCarouselPreviewComponent(elRef) {
        this.elRef = elRef;
        this.class = true;
        this.closedClass = false;
        this.imageUrl = '';
        this.loading = true;
        this.scaleRate = 1;
        this.modalRef = {};
    }
    Object.defineProperty(SliderCarouselPreviewComponent.prototype, "width", {
        get: function () { return (this.loading ? 0 : this.img.width * this.scaleRate) + 'px'; },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SliderCarouselPreviewComponent.prototype, "height", {
        get: function () { return (this.loading ? 0 : this.img.height * this.scaleRate) + 'px'; },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SliderCarouselPreviewComponent.prototype, "windowWidth", {
        get: function () { return window.innerWidth - 40; },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SliderCarouselPreviewComponent.prototype, "windowHeight", {
        get: function () { return window.innerHeight - 40; },
        enumerable: false,
        configurable: true
    });
    SliderCarouselPreviewComponent.prototype.ngOnInit = function () {
        this.imageUrl = this.modalRef.image.safeUrl;
        this.loadImage(this.modalRef.image.pureUrl);
    };
    SliderCarouselPreviewComponent.prototype.loadImage = function (imageUrl) {
        var _this = this;
        this.loading = true;
        this.img = new Image();
        this.img.onload = function () {
            _this.loading = false;
            _this.onWindowResize();
        };
        this.img.src = imageUrl;
    };
    SliderCarouselPreviewComponent.prototype.onImageChange = function (image, fileName) {
        if (!this.loading && this.imageUrl !== image.safeUrl) {
            this.imageUrl = image.safeUrl;
            this.loadImage(image.pureUrl);
        }
    };
    SliderCarouselPreviewComponent.prototype.onKeydown = function (event) {
        event.preventDefault();
        event.stopImmediatePropagation();
        if (event.keyCode === 27)
            this.close();
    };
    SliderCarouselPreviewComponent.prototype.onWindowResize = function () {
        if (!this.loading) {
            var widthDiff = this.windowWidth - this.img.width;
            var heightDiff = this.windowHeight - this.img.height;
            if (widthDiff < 0 || heightDiff < 0) {
                if (widthDiff < heightDiff) {
                    if (this.img.width > this.windowWidth) {
                        this.scaleRate = this.windowWidth / this.img.width;
                        return;
                    }
                }
                else {
                    if (this.img.height > this.windowHeight) {
                        this.scaleRate = this.windowHeight / this.img.height;
                        return;
                    }
                }
            }
        }
        this.scaleRate = 1;
    };
    SliderCarouselPreviewComponent.prototype.close = function () {
        var _this = this;
        this.closedClass = true;
        setTimeout(function () { return _this.modalRef.close(); }, 500);
    };
    SliderCarouselPreviewComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    SliderCarouselPreviewComponent.decorators = [
        { type: Component, args: [{
                    selector: 'slider-carousel-preview',
                    template: "<section>\n    <img [src]=\"imageUrl\" [style.width]=\"width\" [style.height]=\"height\" [style.opacity]=\"loading ? 0 : 1\" />\n    <div *ngIf=\"!loading\" class=\"slider-carousel-preview-actions\">\n        <button matRipple matTooltipPosition=\"above\" (click)=\"close()\">\n            <svg x=\"0px\" y=\"0px\" viewBox=\"0 0 51.976 51.976\">\n                <path d=\"M44.373,7.603c-10.137-10.137-26.632-10.138-36.77,0c-10.138,10.138-10.137,26.632,0,36.77s26.632,10.138,36.77,0\n                    C54.51,34.235,54.51,17.74,44.373,7.603z M36.241,36.241c-0.781,0.781-2.047,0.781-2.828,0l-7.425-7.425l-7.778,7.778\n                    c-0.781,0.781-2.047,0.781-2.828,0c-0.781-0.781-0.781-2.047,0-2.828l7.778-7.778l-7.425-7.425c-0.781-0.781-0.781-2.048,0-2.828\n                    c0.781-0.781,2.047-0.781,2.828,0l7.425,7.425l7.071-7.071c0.781-0.781,2.047-0.781,2.828,0c0.781,0.781,0.781,2.047,0,2.828\n                    l-7.071,7.071l7.425,7.425C37.022,34.194,37.022,35.46,36.241,36.241z\"/>\n            </svg>\n        </button>\n    </div>\n</section>\n<div *ngIf=\"loading\" class=\"slider-carousel-loading\"></div>",
                    host: {
                        '(document:keydown)': 'onKeydown($event)',
                        '(window:resize)': 'onWindowResize()'
                    }
                },] }
    ];
    SliderCarouselPreviewComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    SliderCarouselPreviewComponent.propDecorators = {
        class: [{ type: HostBinding, args: ['class.slider-carousel-preview',] }],
        closedClass: [{ type: HostBinding, args: ['class.closed',] }]
    };
    return SliderCarouselPreviewComponent;
}());
export { SliderCarouselPreviewComponent };
